### ISSUE TITLE
Allow mounting configmaps and secrets

### ISSUE THREAD

**Issue by ctron** [keycloak/keycloak#12722]
### Description

Using the operator, it is not (officially) supported to mount configmaps or secrets.

Assuming you want to use e.g. TLS with Postgres, you might need a certificate for the JDBC driver. However, certificate needs to be present in the file system of the keycloak server.

One can do this by mounting a configMap or secret into the pod. However, as the deployment is fully controlled by the keycloak operator, there is currently no official way.

### Motivation

Using TLS is always good. Using `unsupported` is discouraged.

**Comment by froli-code** [comment]
This seems like a good enhancement to me. K8S provides an API for configmaps and secrets, which could be implemented in the operator.

**Comment by vmuzikar** [comment]
We should rather provider a first class support for configuring truststores. This is covered by https://github.com/keycloak/keycloak/issues/9784

**Comment by vmuzikar** [comment]
@ctron Going to close this in favor of https://github.com/keycloak/keycloak/issues/9784. Let me know if that doesn't work for you.

**Comment by vmuzikar** [comment]
Duplicate of https://github.com/keycloak/keycloak/issues/9784

**Comment by ctron** [comment]
I don't think it does. As you can see from the postgres docs, there are all kinds of paths (and files) which you can configure. But with the Keycloak CRD, you don't have any chance to actually provide those files: https://jdbc.postgresql.org/documentation/use/#connection-parameters/

**Comment by vmuzikar** [comment]
@ctron Ok, let's keep this open for now. But before implementing this in too generic way (which might not be necessary), I'd like to explore if it'd perhaps make sense to add first class support for the few selected parameters.

**Comment by shawkins** [comment]
The trust part of this request should be satisfied by #24798

**Comment by shawkins** [comment]
Closing as a duplicate of #44276 since it's already associated with the unsupported template parent.
